Allow "readonly" as alias to "ro" in mount options

mariushoch opened the pull request:
Apparently docker supports this as well, see [Use a read-only volume](https://docs.docker.com/storage/volumes/#use-a-read-only-volume) in the docker docs.

#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:
To be fully compatible with docker in this regard.

#### How to verify it

With a mount option like `type=bind,source=/var/source,target=/src,readonly`.

```release-note
None
```

giuseppe commented:
bors retry

mariushoch commented:
I just changed the behavior in "ValidateVolumeOpts" as I realized that right now it does not allow multiple/ redundant entries for `ro` (and others), so why should it allow both `readonly` and `ro`?

rhatdan commented:
LGTM

rhatdan commented:
bors r+
Thanks @mariushoch

rhatdan commented:
bors retry

rhatdan commented:
@mariushoch Could you rebase and repush this PR,  we had a fix added for CI to fix a change in the podman packages.

mariushoch commented:
Rebased and reverted my changes to `ValidateVolumeOpts`, it should not allow "readonly" as it is supposed to get normalized volume options only (and storage code wont understand "readonly").

rhatdan commented:
@mariushoch Could you rebase and repush.

rhatdan commented:
bors r+
